sbin/spark-daemon.sh

```diff
@@ -74,10 +75,13 @@ shift
 spark_rotate_log ()
 {
     log=$1;
-    num=5;
-    if [ -n "$2" ]; then
-	num=$2
+
+    if [[ ${SPARK_LOG_MAX_FILES} -gt 5 ]] 2>/dev/null; then
```

Comment:
why do we redirect to dev/null?

Reply:
try to deal with non-digit silently

Reply:
Let's validate if this is a number and show a better error message.

Reply:
OK